pkg/cover/server.go

```diff
@@ -120,11 +129,61 @@ func registerService(c *gin.Context) {
 }
 
 func profile(c *gin.Context) {
-	svrsUnderTest := DefaultStore.GetAll()
+	respByte, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+	var param ProfileParam
+	json.Unmarshal(respByte, &param)
+	if param.Name != "" && param.Address != "" {
+		c.JSON(http.StatusExpectationFailed, gin.H{"error": "invalid param"})
+		return
+	}
+	nameList := strings.Split(param.Name, "&")
+	addrList := strings.Split(param.Address, "&")
+	svrsAll := DefaultStore.GetAll()
+	svrsUnderTest := make(map[string][]string)
+	if param.Name == "" && param.Address == "" {
+		svrsUnderTest = svrsAll
+	} else {
+		if param.Name != "" {
+			for _, name := range nameList {
+				miss := true
+				for svr, addrs := range svrsAll {
+					if svr == name {
+						svrsUnderTest[svr] = addrs
```

Comment:
How to handle duplicate service name?

Reply:
if service name is same as before, the map will assign to the struct.

Reply:
that's the issue, 
1. if different binary share the same name, how? (the filepath is same, import path is different)
2. if same binary has different services depends on args (like goc), how? (both import path and filepath are same)